### PR TITLE
fix parsing of AMQP URL Arg.  Add tracing for AMQP transport connecti…

### PR DESCRIPTION
**Pull request by atyronesmith**
…ons.

### CODE REVIEW

File: bridge.h [hunk 1]
```diff
@@ -24,7 +24,7 @@
 #define DEFAULT_RING_BUFFER_COUNT "5000"
 #define DEFAULT_RING_BUFFER_SIZE "2048"
 
-#define AMQP_URL_REGEX "^amqp://(([a-z]+)(:([a-z]+))*@)*([a-zA-Z_0-9.-]+)(:([0-9]+))*(.+)$"
+#define AMQP_URL_REGEX "^(amqps*)://(([a-z]+)(:([a-z]+))*@)*([a-zA-Z_0-9.-]+)(:([0-9]+))*(.+)$"
```
Comment (csibbitt) [inline bridge.h:27]: No other changes required for amqps?

Comment (leifmadsen) [reply bridge.h:27]: @atyronesmith ping for response?

File: bridge.h [hunk 2]
```diff
@@ -24,7 +24,7 @@
 #define DEFAULT_RING_BUFFER_COUNT "5000"
 #define DEFAULT_RING_BUFFER_SIZE "2048"
 
-#define AMQP_URL_REGEX \
+#define AMQP_URL_REGEX                                                         \
```
Comment (csibbitt) [inline bridge.h:27]: Lol. Sure bot, whatever

**Comment by csibbitt** [comment]
@vyzigold Can you take a look at this and LMK if it conflicts with anything you're working on? I'd like to merge it since it seems helpful, but I don't want to cause you any surprises.

**Comment by vyzigold** [comment]
@csibbitt I don't know what "LMK" means "Let Me Know"? But I didn't want to start working on anything new before going on vacation (so I didn't start working on sg-bridge yet). So this doesn't conflict with anything I'm currently working on and I don't think this will affect anything I'll work on next week, so I don't mind you merging this.

Just out of curiosity, @atyronesmith do you remember, why you needed to add the volatile keyword to the variables in bridge.h? I remember from my C lectures, that volatile does "almost nothing". Is it because the variables may get changed in different thread, than the one that's printing them, so you want to specify, that they may not be the same like in the last iteration?

**Comment by atyronesmith** [comment]
If I remember correctly, I added volatile to address issues with using compiler optimization.  The optimizer will sometimes remove a read from a variable in a loop that it doesn't see changing.  The volatile keyword says that the variable can change outside of what the optimizer can see.

**Pull request merged**